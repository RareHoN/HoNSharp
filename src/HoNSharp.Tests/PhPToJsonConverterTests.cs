using ApprovalTests;
using ApprovalTests.Reporters;
using HoNSharp.Converters;
using Newtonsoft.Json;

namespace HoNSharp.Tests
{
    [UseReporter(typeof(DiffReporter))]
    public class PhPToJsonConverterTests
    {
        private readonly PhPToJsonConverter _converter;

        public PhPToJsonConverterTests()
        {
            _converter = new PhPToJsonConverter();
        }

        [Fact]
        public void ConvertToJson_GetAccountAllHeroStats_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_account_all_hero_stats_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_GetMatchStats_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_match_stats_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_GetMatchStats_WithDateTimeOffsetError_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_match_stats_datetimeoffset_error_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_GetMatchStats_WithUnexpectedCharacterError_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("get_match_stats_unexpected_character_error_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_MatchHistoryOverviewMidwars_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("match_history_overview_midwars_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_MatchHistoryOverviewNormal_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("match_history_overview_normal_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_ShowStatsCampaign_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("show_stats_campaign_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_ShowStatsCampaignMidwars_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("show_stats_campaign_midwars_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_ShowStatsMastery_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("show_stats_mastery_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        [Fact]
        public void ConvertToJson_SimpleStats_ShouldCorrectlyConvert()
        {
            // Arrange
            var responseFileContent = ReadFile("simple_stats_response.txt");

            // Act
            var result = _converter.ConvertToJson(responseFileContent);

            // Assert
            Approvals.Verify(FormatJson(result));
        }

        private string ReadFile(string responseFileName)
        {
            var responseFilePath = Path.Combine(Directory.GetCurrentDirectory(), "TestData", responseFileName);

            return File.ReadAllText(responseFilePath);
        }

        private string FormatJson(string json)
        {
            var deserialized = JsonConvert.DeserializeObject(json);
            return JsonConvert.SerializeObject(deserialized, Formatting.Indented);
        }
    }
}