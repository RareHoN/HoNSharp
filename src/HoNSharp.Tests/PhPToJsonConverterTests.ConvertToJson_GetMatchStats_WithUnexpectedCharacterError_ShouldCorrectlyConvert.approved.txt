{
  "match_summ": {
    "507444": {
      "match_id": 507444,
      "map": "caldavar",
      "time_played": 2542,
      "version": "4.10.1.8",
      "date": "11/7/2022",
      "time": "22:44:08.6404153",
      "mname": null,
      "class": null,
      "nm": 0,
      "sd": 0,
      "rd": 0,
      "dm": 0,
      "bd": 0,
      "bp": 0,
      "ar": 0,
      "cm": 0,
      "lp": 0,
      "bb": 0,
      "bm": 0,
      "km": 0,
      "alt_pick": 0,
      "shuf": 0,
      "ap": 0,
      "br": 0,
      "em": 0,
      "cas": 0,
      "nl": 0,
      "no_stats": 0,
      "ab": 0,
      "verified_only": 0,
      "gated": 0,
      "timestamp": 1667861048,
      "url": null,
      "size": 1,
      "name": null,
      "s3_url": "http://api.kongor.online/replays/M507444.honreplay",
      "winning_team": 2,
      "gamemode": "reborn",
      "mvp": 57578,
      "awd_mann": "-1",
      "awd_mqk": "-1",
      "awd_lgks": 24971,
      "awd_msd": 57578,
      "awd_mkill": 57578,
      "awd_masst": 50821,
      "awd_ledth": 24971,
      "awd_mbdmg": 24971,
      "awd_mwk": "-1",
      "awd_mhdd": 83079,
      "awd_hcs": 83079,
      "mmpoints": 7
    }
  },
  "match_player_stats": {
    "507444": {
      "0": {
        "nickname": "[K]ONGOR",
        "clan_tag": "K",
        "clan_id": 1,
        "team": 1,
        "position": 4,
        "group_num": 3,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 0,
        "losses": 1,
        "discos": 0,
        "concedes": 1,
        "kicked": 0,
        "social_bonus": 2,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": -12.4643,
        "amm_team_count": 1,
        "concedevotes": 0,
        "herokills": 1,
        "herodmg": 4514,
        "herokillsgold": 1520,
        "heroassists": 7,
        "heroexp": 2871,
        "deaths": 14,
        "buybacks": 0,
        "goldlost2death": 2491,
        "secs_dead": 503,
        "teamcreepkills": 46,
        "teamcreepdmg": 19312,
        "teamcreepgold": 1820,
        "teamcreepexp": 5232,
        "neutralcreepkills": 63,
        "neutralcreepdmg": 32871,
        "neutralcreepgold": 2089,
        "neutralcreepexp": 4149,
        "bdmg": 25,
        "razed": 0,
        "bdmgexp": 0,
        "bgold": 800,
        "denies": 7,
        "exp_denied": 329,
        "gold": 6741,
        "gold_spent": 6340,
        "exp": 12254,
        "actions": 3424,
        "secs": 2542,
        "level": 15,
        "consumables": 12,
        "wards": 4,
        "bloodlust": 0,
        "doublekill": 0,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 0,
        "ks4": 0,
        "ks5": 0,
        "ks6": 0,
        "ks7": 0,
        "ks8": 0,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 0,
        "humiliation": 0,
        "nemesis": 0,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 0,
        "gameplaystat1": 0,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2542,
        "match_id": 507444,
        "account_id": 1,
        "map": "caldavar",
        "cli_name": "Hero_Tempest",
        "mdt": "11/7/2022"
      },
      "1": {
        "nickname": "_Mikhe_",
        "clan_tag": null,
        "clan_id": 0,
        "team": 2,
        "position": 5,
        "group_num": 0,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 1,
        "losses": 0,
        "discos": 0,
        "concedes": 0,
        "kicked": 0,
        "social_bonus": 1,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": 4.9229,
        "amm_team_count": 1,
        "concedevotes": 0,
        "herokills": 9,
        "herodmg": 9169,
        "herokillsgold": 6252,
        "heroassists": 5,
        "heroexp": 12300,
        "deaths": 4,
        "buybacks": 0,
        "goldlost2death": 1290,
        "secs_dead": 110,
        "teamcreepkills": 23,
        "teamcreepdmg": 16144,
        "teamcreepgold": 977,
        "teamcreepexp": 5254,
        "neutralcreepkills": 6,
        "neutralcreepdmg": 5017,
        "neutralcreepgold": 485,
        "neutralcreepexp": 1304,
        "bdmg": 177,
        "razed": 0,
        "bdmgexp": 0,
        "bgold": 3650,
        "denies": 2,
        "exp_denied": 68,
        "gold": 13695,
        "gold_spent": 12795,
        "exp": 18897,
        "actions": 5323,
        "secs": 2542,
        "level": 19,
        "consumables": 16,
        "wards": 13,
        "bloodlust": 1,
        "doublekill": 0,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 1,
        "ks4": 1,
        "ks5": 0,
        "ks6": 0,
        "ks7": 0,
        "ks8": 0,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 0,
        "humiliation": 0,
        "nemesis": 0,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 0,
        "gameplaystat1": 0,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2542,
        "match_id": 507444,
        "account_id": 63887,
        "map": "caldavar",
        "cli_name": "Hero_Engineer",
        "mdt": "11/7/2022"
      },
      "2": {
        "nickname": "crazyboi",
        "clan_tag": null,
        "clan_id": 0,
        "team": 1,
        "position": 3,
        "group_num": 3,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 0,
        "losses": 1,
        "discos": 0,
        "concedes": 1,
        "kicked": 0,
        "social_bonus": 2,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": -11.0478,
        "amm_team_count": 1,
        "concedevotes": 1,
        "herokills": 6,
        "herodmg": 14498,
        "herokillsgold": 3331,
        "heroassists": 5,
        "heroexp": 6677,
        "deaths": 5,
        "buybacks": 0,
        "goldlost2death": 2850,
        "secs_dead": 300,
        "teamcreepkills": 236,
        "teamcreepdmg": 104680,
        "teamcreepgold": 10485,
        "teamcreepexp": 12216,
        "neutralcreepkills": 122,
        "neutralcreepdmg": 100093,
        "neutralcreepgold": 5059,
        "neutralcreepexp": 7293,
        "bdmg": 495,
        "razed": 1,
        "bdmgexp": 0,
        "bgold": 1010,
        "denies": 10,
        "exp_denied": 347,
        "gold": 19938,
        "gold_spent": 20020,
        "exp": 26188,
        "actions": 5148,
        "secs": 2542,
        "level": 22,
        "consumables": 19,
        "wards": 0,
        "bloodlust": 0,
        "doublekill": 1,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 1,
        "ks4": 1,
        "ks5": 1,
        "ks6": 0,
        "ks7": 0,
        "ks8": 0,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 0,
        "humiliation": 0,
        "nemesis": 0,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 0,
        "gameplaystat1": 0,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2542,
        "match_id": 507444,
        "account_id": 510,
        "map": "caldavar",
        "cli_name": "Hero_Dampeer",
        "mdt": "11/7/2022"
      },
      "3": {
        "nickname": "dashorrorkid",
        "clan_tag": null,
        "clan_id": 0,
        "team": 2,
        "position": 8,
        "group_num": 3,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 1,
        "losses": 0,
        "discos": 0,
        "concedes": 0,
        "kicked": 0,
        "social_bonus": 1,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": 13.9373,
        "amm_team_count": 1,
        "concedevotes": 0,
        "herokills": 2,
        "herodmg": 4647,
        "herokillsgold": 2322,
        "heroassists": 8,
        "heroexp": 7091,
        "deaths": 3,
        "buybacks": 0,
        "goldlost2death": 1044,
        "secs_dead": 162,
        "teamcreepkills": 104,
        "teamcreepdmg": 63629,
        "teamcreepgold": 4699,
        "teamcreepexp": 7659,
        "neutralcreepkills": 273,
        "neutralcreepdmg": 234233,
        "neutralcreepgold": 10100,
        "neutralcreepexp": 17085,
        "bdmg": 4670,
        "razed": 2,
        "bdmgexp": 0,
        "bgold": 3910,
        "denies": 2,
        "exp_denied": 150,
        "gold": 23282,
        "gold_spent": 19510,
        "exp": 31895,
        "actions": 4032,
        "secs": 2542,
        "level": 24,
        "consumables": 16,
        "wards": 0,
        "bloodlust": 0,
        "doublekill": 0,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 0,
        "ks4": 0,
        "ks5": 0,
        "ks6": 0,
        "ks7": 0,
        "ks8": 0,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 0,
        "humiliation": 0,
        "nemesis": 0,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 0,
        "gameplaystat1": 0,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2542,
        "match_id": 507444,
        "account_id": 587,
        "map": "caldavar",
        "cli_name": "Hero_Solstice",
        "mdt": "11/7/2022"
      },
      "4": {
        "nickname": "Dilonusikiym",
        "clan_tag": null,
        "clan_id": 0,
        "team": 1,
        "position": 0,
        "group_num": 0,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 0,
        "losses": 1,
        "discos": 0,
        "concedes": 1,
        "kicked": 0,
        "social_bonus": 1,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": -17.3933,
        "amm_team_count": 1,
        "concedevotes": 1,
        "herokills": 6,
        "herodmg": 9080,
        "herokillsgold": 3503,
        "heroassists": 7,
        "heroexp": 7678,
        "deaths": 8,
        "buybacks": 0,
        "goldlost2death": 2516,
        "secs_dead": 318,
        "teamcreepkills": 124,
        "teamcreepdmg": 44268,
        "teamcreepgold": 5588,
        "teamcreepexp": 10958,
        "neutralcreepkills": 12,
        "neutralcreepdmg": 10402,
        "neutralcreepgold": 700,
        "neutralcreepexp": 1363,
        "bdmg": 373,
        "razed": 0,
        "bdmgexp": 0,
        "bgold": 800,
        "denies": 4,
        "exp_denied": 147,
        "gold": 10978,
        "gold_spent": 11400,
        "exp": 20001,
        "actions": 7716,
        "secs": 2542,
        "level": 19,
        "consumables": 28,
        "wards": 0,
        "bloodlust": 0,
        "doublekill": 1,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 0,
        "ks4": 0,
        "ks5": 0,
        "ks6": 0,
        "ks7": 0,
        "ks8": 0,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 1,
        "humiliation": 0,
        "nemesis": 0,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 35,
        "gameplaystat1": 16,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2542,
        "match_id": 507444,
        "account_id": 57578,
        "map": "caldavar",
        "cli_name": "Hero_Devourer",
        "mdt": "11/7/2022"
      },
      "5": {
        "nickname": "Gsson",
        "clan_tag": null,
        "clan_id": 0,
        "team": 2,
        "position": 9,
        "group_num": 4,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 1,
        "losses": 0,
        "discos": 0,
        "concedes": 0,
        "kicked": 0,
        "social_bonus": 1,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": 15.6936,
        "amm_team_count": 1,
        "concedevotes": 0,
        "herokills": 5,
        "herodmg": 13177,
        "herokillsgold": 3565,
        "heroassists": 18,
        "heroexp": 9697,
        "deaths": 7,
        "buybacks": 1,
        "goldlost2death": 2910,
        "secs_dead": 335,
        "teamcreepkills": 110,
        "teamcreepdmg": 30011,
        "teamcreepgold": 4971,
        "teamcreepexp": 9069,
        "neutralcreepkills": 13,
        "neutralcreepdmg": 8322,
        "neutralcreepgold": 598,
        "neutralcreepexp": 1182,
        "bdmg": 754,
        "razed": 2,
        "bdmgexp": 0,
        "bgold": 3910,
        "denies": 4,
        "exp_denied": 248,
        "gold": 13037,
        "gold_spent": 10325,
        "exp": 19951,
        "actions": 6250,
        "secs": 2542,
        "level": 19,
        "consumables": 20,
        "wards": 1,
        "bloodlust": 0,
        "doublekill": 0,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 1,
        "ks4": 0,
        "ks5": 0,
        "ks6": 0,
        "ks7": 0,
        "ks8": 0,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 0,
        "humiliation": 0,
        "nemesis": 0,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 0,
        "gameplaystat1": 0,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2542,
        "match_id": 507444,
        "account_id": 50821,
        "map": "caldavar",
        "cli_name": "Hero_Grinex",
        "mdt": "11/7/2022"
      },
      "6": {
        "nickname": "Naphten",
        "clan_tag": null,
        "clan_id": 0,
        "team": 1,
        "position": 1,
        "group_num": 1,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 0,
        "losses": 1,
        "discos": 0,
        "concedes": 1,
        "kicked": 0,
        "social_bonus": 1,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": -5.8117,
        "amm_team_count": 1,
        "concedevotes": 0,
        "herokills": 4,
        "herodmg": 8514,
        "herokillsgold": 2625,
        "heroassists": 5,
        "heroexp": 6308,
        "deaths": 6,
        "buybacks": 0,
        "goldlost2death": 2345,
        "secs_dead": 262,
        "teamcreepkills": 180,
        "teamcreepdmg": 85732,
        "teamcreepgold": 8229,
        "teamcreepexp": 11997,
        "neutralcreepkills": 87,
        "neutralcreepdmg": 69802,
        "neutralcreepgold": 3201,
        "neutralcreepexp": 5341,
        "bdmg": 263,
        "razed": 1,
        "bdmgexp": 0,
        "bgold": 1010,
        "denies": 4,
        "exp_denied": 150,
        "gold": 15773,
        "gold_spent": 16300,
        "exp": 23648,
        "actions": 5177,
        "secs": 2542,
        "level": 23,
        "consumables": 14,
        "wards": 0,
        "bloodlust": 0,
        "doublekill": 0,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 0,
        "ks4": 0,
        "ks5": 0,
        "ks6": 0,
        "ks7": 0,
        "ks8": 0,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 0,
        "humiliation": 0,
        "nemesis": 0,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 0,
        "gameplaystat1": 0,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2542,
        "match_id": 507444,
        "account_id": 39914,
        "map": "caldavar",
        "cli_name": "Hero_MonkeyKing",
        "mdt": "11/7/2022"
      },
      "7": {
        "nickname": "RareHoN",
        "clan_tag": null,
        "clan_id": 0,
        "team": 2,
        "position": 6,
        "group_num": 1,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 1,
        "losses": 0,
        "discos": 0,
        "concedes": 0,
        "kicked": 0,
        "social_bonus": 1,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": 5.4825,
        "amm_team_count": 1,
        "concedevotes": 0,
        "herokills": 11,
        "herodmg": 22865,
        "herokillsgold": 5115,
        "heroassists": 9,
        "heroexp": 10697,
        "deaths": 4,
        "buybacks": 0,
        "goldlost2death": 2130,
        "secs_dead": 290,
        "teamcreepkills": 277,
        "teamcreepdmg": 132919,
        "teamcreepgold": 11746,
        "teamcreepexp": 13519,
        "neutralcreepkills": 129,
        "neutralcreepdmg": 102557,
        "neutralcreepgold": 4878,
        "neutralcreepexp": 8044,
        "bdmg": 5467,
        "razed": 4,
        "bdmgexp": 0,
        "bgold": 4020,
        "denies": 24,
        "exp_denied": 1082,
        "gold": 27432,
        "gold_spent": 28865,
        "exp": 32347,
        "actions": 9310,
        "secs": 2542,
        "level": 25,
        "consumables": 16,
        "wards": 0,
        "bloodlust": 0,
        "doublekill": 1,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 1,
        "ks4": 1,
        "ks5": 0,
        "ks6": 0,
        "ks7": 0,
        "ks8": 0,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 0,
        "humiliation": 0,
        "nemesis": 98,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 0,
        "gameplaystat1": 0,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2436,
        "match_id": 507444,
        "account_id": 83079,
        "map": "caldavar",
        "cli_name": "Hero_Nomad",
        "mdt": "11/7/2022"
      },
      "8": {
        "nickname": "Suczuk",
        "clan_tag": null,
        "clan_id": 0,
        "team": 2,
        "position": 7,
        "group_num": 2,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 1,
        "losses": 0,
        "discos": 0,
        "concedes": 0,
        "kicked": 0,
        "social_bonus": 1,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": 11.1612,
        "amm_team_count": 1,
        "concedevotes": 0,
        "herokills": 14,
        "herodmg": 22580,
        "herokillsgold": 6404,
        "heroassists": 11,
        "heroexp": 13449,
        "deaths": 3,
        "buybacks": 0,
        "goldlost2death": 960,
        "secs_dead": 132,
        "teamcreepkills": 221,
        "teamcreepdmg": 109469,
        "teamcreepgold": 9793,
        "teamcreepexp": 14726,
        "neutralcreepkills": 22,
        "neutralcreepdmg": 23822,
        "neutralcreepgold": 1186,
        "neutralcreepexp": 2346,
        "bdmg": 7674,
        "razed": 8,
        "bdmgexp": 0,
        "bgold": 4690,
        "denies": 8,
        "exp_denied": 370,
        "gold": 24048,
        "gold_spent": 21625,
        "exp": 30554,
        "actions": 5150,
        "secs": 2542,
        "level": 24,
        "consumables": 10,
        "wards": 2,
        "bloodlust": 0,
        "doublekill": 1,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 1,
        "ks4": 1,
        "ks5": 1,
        "ks6": 1,
        "ks7": 1,
        "ks8": 1,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 0,
        "humiliation": 0,
        "nemesis": 38,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 0,
        "gameplaystat1": 0,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2542,
        "match_id": 507444,
        "account_id": 24971,
        "map": "caldavar",
        "cli_name": "Hero_FlintBeastwood",
        "mdt": "11/7/2022"
      },
      "9": {
        "nickname": "tOnizhitO_16",
        "clan_tag": null,
        "clan_id": 0,
        "team": 1,
        "position": 2,
        "group_num": 2,
        "benefit": 0,
        "hero_id": 4294967295,
        "wins": 0,
        "losses": 1,
        "discos": 0,
        "concedes": 1,
        "kicked": 0,
        "social_bonus": 1,
        "used_token": 0,
        "pub_skill": 0,
        "pub_count": 0,
        "amm_team_rating": -8.0873,
        "amm_team_count": 1,
        "concedevotes": 4,
        "herokills": 4,
        "herodmg": 5999,
        "herokillsgold": 5453,
        "heroassists": 5,
        "heroexp": 5494,
        "deaths": 8,
        "buybacks": 0,
        "goldlost2death": 2043,
        "secs_dead": 350,
        "teamcreepkills": 22,
        "teamcreepdmg": 11526,
        "teamcreepgold": 858,
        "teamcreepexp": 6114,
        "neutralcreepkills": 12,
        "neutralcreepdmg": 5675,
        "neutralcreepgold": 616,
        "neutralcreepexp": 2117,
        "bdmg": 270,
        "razed": 0,
        "bdmgexp": 0,
        "bgold": 800,
        "denies": 3,
        "exp_denied": 53,
        "gold": 9137,
        "gold_spent": 9605,
        "exp": 13727,
        "actions": 5508,
        "secs": 2542,
        "level": 16,
        "consumables": 12,
        "wards": 13,
        "bloodlust": 0,
        "doublekill": 0,
        "triplekill": 0,
        "quadkill": 0,
        "annihilation": 0,
        "ks3": 0,
        "ks4": 0,
        "ks5": 0,
        "ks6": 0,
        "ks7": 0,
        "ks8": 0,
        "ks9": 0,
        "ks10": 0,
        "ks15": 0,
        "smackdown": 1,
        "humiliation": 0,
        "nemesis": 0,
        "retribution": 0,
        "score": 0,
        "gameplaystat0": 0,
        "gameplaystat1": 0,
        "gameplaystat2": 0,
        "gameplaystat3": 0,
        "gameplaystat4": 0,
        "gameplaystat5": 0,
        "gameplaystat6": 0,
        "gameplaystat7": 0,
        "gameplaystat8": 0,
        "gameplaystat9": 0,
        "time_earning_exp": 2542,
        "match_id": 507444,
        "account_id": 1771,
        "map": "caldavar",
        "cli_name": "Hero_WitchSlayer",
        "mdt": "11/7/2022"
      }
    }
  },
  "inventory": {
    "0": {
      "83079": {
        "slot_1": "Item_PortalKey",
        "slot_2": "Item_Protect",
        "slot_3": "Item_Splash",
        "slot_4": "Item_EnhancedMarchers",
        "slot_5": "Item_Silence",
        "slot_6": "Item_Damage9"
      },
      "57578": {
        "slot_1": "Item_Striders",
        "slot_2": "Item_MagicArmor2",
        "slot_3": "Item_Intelligence7",
        "slot_4": "Item_IronShield",
        "slot_5": "Item_BloodChalice",
        "slot_6": "Item_ShieldOfTheFive"
      },
      "63887": {
        "slot_1": "Item_PortalKey",
        "slot_2": "Item_BloodChalice",
        "slot_5": "Item_Intelligence7",
        "slot_6": "Item_Striders"
      },
      "50821": {
        "slot_1": "Item_EnhancedMarchers",
        "slot_2": "Item_PortalKey",
        "slot_3": "Item_SacrificialStone",
        "slot_4": "Item_IronShield",
        "slot_5": "Item_MysticVestments",
        "slot_6": "Item_Madfreds_BrassKnuckles"
      },
      "1": {
        "slot_1": "Item_ArcaneBoots",
        "slot_3": "Item_PortalKey",
        "slot_4": "Item_WindWhistle"
      },
      "510": {
        "slot_1": "Item_EnhancedMarchers",
        "slot_2": "Item_PortalKey",
        "slot_3": "Item_Stealth",
        "slot_4": "Item_Intelligence7",
        "slot_5": "Item_BloodChalice",
        "slot_6": "Item_Dawnbringer"
      },
      "39914": {
        "slot_1": "Item_EnhancedMarchers",
        "slot_2": "Item_Silence",
        "slot_3": "Item_Sasuke",
        "slot_4": "Item_Madfreds_BrassKnuckles",
        "slot_5": "Item_MysticVestments",
        "slot_6": "Item_LoggersHatchet"
      },
      "587": {
        "slot_1": "Item_DaemonicBreastplate",
        "slot_2": "Item_EnhancedMarchers",
        "slot_4": "Item_ElderParasite",
        "slot_5": "Item_Immunity",
        "slot_6": "Item_Pierce"
      },
      "1771": {
        "slot_1": "Item_GraveLocket",
        "slot_2": "Item_VoidTalisman",
        "slot_3": "Item_Striders",
        "slot_4": "Item_PortalKey"
      },
      "24971": {
        "slot_1": "Item_EnhancedMarchers",
        "slot_2": "Item_Immunity",
        "slot_3": "Item_ElderParasite",
        "slot_4": "Item_Protect",
        "slot_5": "Item_Madfreds_BrassKnuckles",
        "slot_6": "Item_Evasion"
      }
    }
  },
  "0": true
}